src/components/network-map-tab.js

```diff
@@ -225,51 +238,100 @@ export const NetworkMapTab = ({
     ]);
 
-    const handleReloadMap = useCallback(() => {
-        reloadMapGeoData();
-        dispatch(setForceNetworkReload());
-    }, [dispatch, reloadMapGeoData]);
+    const loadMapEquipments = useCallback(() => {
+        if (!isNodeBuilt(currentNode) || !studyUuid) {
+            return;
+        }
+        if (!isInitialized) {
+            const initialMapEquipments = new MapEquipments(
+                studyUuid,
+                currentNode?.id,
+                setErrorMessage,
+                dispatch,
+                intlRef
+            );
+            dispatch(mapEquipmentsCreated(initialMapEquipments));
+        } else {
+            console.info('Reload map equipments');
+            mapEquipments.reloadImpactedSubstationsEquipments(
+                studyUuid,
+                currentNode,
+                updatedSubstationsIds,
+                setUpdatedLines
+            );
+        }
+    }, [
+        currentNode,
+        dispatch,
+        intlRef,
+        isInitialized,
+        mapEquipments,
+        setErrorMessage,
+        studyUuid,
+        updatedSubstationsIds,
+    ]);
+
+    useEffect(() => {
+        if (!mapEquipments || refIsMapManualRefreshEnabled.current) {
+            return;
+        }
+        if (deletedEquipment) {
+            mapEquipments?.removeEquipment(
+                deletedEquipment?.type,
+                deletedEquipment?.id
+            );
+        }
+    }, [deletedEquipment, mapEquipments]);
+
+    const handleFullMapReload = useCallback(() => {
+        loadMapEquipments();
+        loadMapGeoData();
+    }, [loadMapEquipments, loadMapGeoData]);
 
     useEffect(() => {
         let previousCurrentNode = currentNodeRef.current;
         currentNodeRef.current = currentNode;
         // if only renaming, do not reload geo data
         if (isNodeRenamed(previousCurrentNode, currentNode)) return;
         if (disabled) return;
-        if (mapManualRefresh && isInitialized) return;
+        if (refIsMapManualRefreshEnabled.current && isInitialized) return;
         // Hack to avoid reload Geo Data when switching display mode to TREE then back to MAP or HYBRID
         // TODO REMOVE LATER
         if (!reloadMapNeeded) return;
-
-        reloadMapGeoData();
+        handleFullMapReload();
         setInitialized(true);
         // Note: studyUuid and dispatch don't change
     }, [
         disabled,
-        reloadMapNeeded,
         studyUuid,
         currentNode,
-        lineFullPath,
-        setWaitingLoadGeoData,
-        setErrorMessage,
-        setGeoData,
-        intlRef,
-        mapManualRefresh,
+        handleFullMapReload,
         isInitialized,
-        reloadMapGeoData,
+        reloadMapNeeded,
+        updatedSubstationsIds,
     ]);
 
+    useEffect(() => {
+        let previousNodeStatus = isCurrentNodeBuiltRef.current;
+        isCurrentNodeBuiltRef.current = isNodeBuilt(currentNode);
+
+        //when we build node we want the map to be up to date
```

Comment:
This loadMap will not be triggered when passing from a built node to another one. Is it ok ?

Reply:
I guess it is not an issue since loadMapEquipments is still triggered in automatic mode when passing from a built node to another one and in manual mode we explicitely want the user to reload the map when he is switching nodes